src/Disks/VolumeJBOD.cpp

```diff
@@ -40,20 +40,28 @@ VolumeJBOD::VolumeJBOD(
         auto ratio = config.getDouble(config_prefix + ".max_data_part_size_ratio");
         if (ratio < 0)
             throw Exception(ErrorCodes::EXCESSIVE_ELEMENT_IN_CONFIG, "'max_data_part_size_ratio' have to be not less then 0.");
+
         UInt64 sum_size = 0;
         std::vector<UInt64> sizes;
         for (const auto & disk : disks)
         {
-            sizes.push_back(disk->getTotalSpace());
-            sum_size += sizes.back();
+            auto size = disk->getTotalSpace();
+            sizes.push_back(*size);
+            if (size)
+                sum_size += *size;
+            else
+                break;
```

Comment:
Push should go after check
```suggestion
            auto size = disk->getTotalSpace();
            if (size)
                sum_size += *size;
            else
                break;
            sizes.push_back(*size);
```